For convenience, here's the result of the tests on my Ubuntu 14:
```console
$ spack test
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.6, pytest-3.0.5, py-1.4.32, pluggy-0.4.0
rootdir: /home/mculpo/PycharmProjects/spack, inifile: pytest.ini
plugins: xdist-1.15.0, cov-2.4.0
collected 531 items 

lib/spack/spack/test/architecture.py .......
lib/spack/spack/test/build_system_guess.py .........
lib/spack/spack/test/build_systems.py .
lib/spack/spack/test/cc.py ................
lib/spack/spack/test/compilers.py ..
lib/spack/spack/test/concretize.py ...................................F..F......
lib/spack/spack/test/concretize_preferences.py .......F
lib/spack/spack/test/config.py ............
lib/spack/spack/test/database.py ....FFF..FF....F
lib/spack/spack/test/directory_layout.py ...
lib/spack/spack/test/environment.py .........
lib/spack/spack/test/file_cache.py ..
lib/spack/spack/test/git_fetch.py ....
lib/spack/spack/test/graph.py ....
lib/spack/spack/test/hg_fetch.py ..
lib/spack/spack/test/install.py ...
lib/spack/spack/test/library_list.py ......
lib/spack/spack/test/link_tree.py ....
lib/spack/spack/test/lock.py ....................................
lib/spack/spack/test/make_executable.py ......
lib/spack/spack/test/mirror.py .....
lib/spack/spack/test/modules.py ................
lib/spack/spack/test/multimethod.py ..........
lib/spack/spack/test/namespace_trie.py .....
lib/spack/spack/test/optional_deps.py ..................
lib/spack/spack/test/package_sanity.py ....
lib/spack/spack/test/packages.py ............
lib/spack/spack/test/pattern.py ...
lib/spack/spack/test/provider_index.py .....
lib/spack/spack/test/python_version.py ..
lib/spack/spack/test/sbang.py ..
lib/spack/spack/test/spack_yaml.py ....
lib/spack/spack/test/spec_dag.py ......................................
lib/spack/spack/test/spec_semantics.py ...................................
lib/spack/spack/test/spec_syntax.py ..............................
lib/spack/spack/test/spec_yaml.py ........
lib/spack/spack/test/stage.py ................
lib/spack/spack/test/svn_fetch.py ..
lib/spack/spack/test/url_extrapolate.py ........
lib/spack/spack/test/url_parse.py ...........................................................
lib/spack/spack/test/url_substitution.py ..
lib/spack/spack/test/versions.py .................................
lib/spack/spack/test/cmd/find.py .
lib/spack/spack/test/cmd/install.py ..
lib/spack/spack/test/cmd/module.py ......
lib/spack/spack/test/cmd/python.py .
lib/spack/spack/test/cmd/test_compiler_cmd.py ..
lib/spack/spack/test/cmd/uninstall.py .
lib/spack/spack/test/cmd/url.py ...x..
============================================================================================= short test summary info =============================================================================================
FAIL lib/spack/spack/test/concretize.py::TestConcretize::()::test_external_package
FAIL lib/spack/spack/test/concretize.py::TestConcretize::()::test_external_and_virtual
FAIL lib/spack/spack/test/concretize_preferences.py::TestConcretizePreferences::()::test_external_mpi
FAIL lib/spack/spack/test/database.py::test_020_db_sanity
FAIL lib/spack/spack/test/database.py::test_025_reindex
FAIL lib/spack/spack/test/database.py::test_030_db_sanity_from_another_process
FAIL lib/spack/spack/test/database.py::test_060_remove_and_add_root_package
FAIL lib/spack/spack/test/database.py::test_070_remove_and_add_dependency_package
FAIL lib/spack/spack/test/database.py::test_external_entries_in_db
XFAIL lib/spack/spack/test/cmd/url.py::test_url_parse_xfail

============================================================================================ slowest 20 test durations ============================================================================================
58.47s call     lib/spack/spack/test/modules.py::TestTcl::test_autoload
33.27s call     lib/spack/spack/test/modules.py::TestTcl::test_prerequisites
32.71s call     lib/spack/spack/test/modules.py::TestLmod::test_autoload
23.11s call     lib/spack/spack/test/modules.py::TestTcl::test_blacklist
13.32s call     lib/spack/spack/test/modules.py::TestDotkit::test_dotkit
10.80s call     lib/spack/spack/test/modules.py::TestLmod::test_blacklist
8.38s call     lib/spack/spack/test/modules.py::TestLmod::test_alter_environment
7.40s call     lib/spack/spack/test/modules.py::TestTcl::test_alter_environment
7.23s call     lib/spack/spack/test/modules.py::TestTcl::test_conflicts
7.09s call     lib/spack/spack/test/modules.py::TestTcl::test_setup_environment
7.01s setup    lib/spack/spack/test/cmd/uninstall.py::test_uninstall
6.49s setup    lib/spack/spack/test/cmd/module.py::test_exit_with_failure[failure_args0]
6.43s setup    lib/spack/spack/test/spec_syntax.py::TestSpecSyntax::test_spec_by_hash
5.76s call     lib/spack/spack/test/python_version.py::PythonVersionTest::test_core_module_compatibility
5.01s setup    lib/spack/spack/test/database.py::test_default_queries
4.74s teardown lib/spack/spack/test/database.py::test_030_db_sanity_from_another_process
4.22s call     lib/spack/spack/test/cmd/url.py::test_url_list
3.48s call     lib/spack/spack/test/modules.py::TestLmod::test_simple_case
3.41s call     lib/spack/spack/test/modules.py::TestTcl::test_simple_case
3.35s call     lib/spack/spack/test/python_version.py::PythonVersionTest::test_package_module_compatibility
==================================================================================================== FAILURES =====================================================================================================
...
================================================================================ 9 failed, 521 passed, 1 xfailed in 293.25 seconds ================================================================================
```

--- a/lib/spack/spack/spec.py
+++ b/lib/spack/spack/spec.py
@@ -3009,7 +3009,6 @@ class SpecParser(spack.parse.Parser):
         spec.variants = VariantMap(spec)
         spec.architecture = None
         spec.compiler = None
-        spec.external = False
         spec.external_path = None
         spec.external_module = None
         spec.compiler_flags = FlagMap(spec)

--- a/lib/spack/spack/test/concretize.py
+++ b/lib/spack/spack/test/concretize.py
@@ -276,7 +276,7 @@ class TestConcretize(object):
     def test_external_package(self):
         spec = Spec('externaltool%gcc')
         spec.concretize()
-        assert spec['externaltool'].external == '/path/to/external_tool'
+        assert spec['externaltool'].external_path == '/path/to/external_tool'
         assert 'externalprereq' not in spec
         assert spec['externaltool'].compiler.satisfies('gcc')
 
@@ -305,8 +305,8 @@ class TestConcretize(object):
     def test_external_and_virtual(self):
         spec = Spec('externaltest')
         spec.concretize()
-        assert spec['externaltool'].external == '/path/to/external_tool'
-        assert spec['stuff'].external == '/path/to/external_virtual_gcc'
+        assert spec['externaltool'].external_path == '/path/to/external_tool'
+        assert spec['stuff'].external_path == '/path/to/external_virtual_gcc'
         assert spec['externaltool'].compiler.satisfies('gcc')
         assert spec['stuff'].compiler.satisfies('gcc')
 

--- a/lib/spack/spack/test/concretize_preferences.py
+++ b/lib/spack/spack/test/concretize_preferences.py
@@ -170,4 +170,4 @@ mpich:
         # ensure that once config is in place, external is used
         spec = Spec('mpi')
         spec.concretize()
-        assert spec['mpich'].external == '/dummy/path'
+        assert spec['mpich'].external_path == '/dummy/path'

--- a/lib/spack/spack/test/database.py
+++ b/lib/spack/spack/test/database.py
@@ -89,12 +89,12 @@ def _check_db_sanity(install_db):
     pkg_in_layout = sorted(spack.store.layout.all_specs())
     actual = sorted(install_db.query())
 
-    externals = sorted([x for x in actual if x.external is not None])
+    externals = sorted([x for x in actual if x.external is True])
     nexpected = len(pkg_in_layout) + len(externals)
 
     assert nexpected == len(actual)
 
-    non_external_in_db = sorted([x for x in actual if x.external is None])
+    non_external_in_db = sorted([x for x in actual if x.external is False])
 
     for e, a in zip(pkg_in_layout, non_external_in_db):
         assert e == a
@@ -377,16 +377,16 @@ def test_external_entries_in_db(database):
     install_db = database.mock.db
 
     rec = install_db.get_record('mpileaks ^zmpi')
-    assert rec.spec.external is None
+    assert rec.spec.external_path is None
     assert rec.spec.external_module is None
 
     rec = install_db.get_record('externaltool')
-    assert rec.spec.external == '/path/to/external_tool'
+    assert rec.spec.external_path == '/path/to/external_tool'
     assert rec.spec.external_module is None
     assert rec.explicit is False
 
     rec.spec.package.do_install(fake=True, explicit=True)
     rec = install_db.get_record('externaltool')
-    assert rec.spec.external == '/path/to/external_tool'
+    assert rec.spec.external_path == '/path/to/external_tool'
     assert rec.spec.external_module is None
     assert rec.explicit is True

